crates/cairo-lint-core/tests/test_files/manual_unwrap_or_default/manual_unwrap_or_default

```diff
@@ -0,0 +1,574 @@
+//! > manual unwrap or default for if let with array!
+
+//! > cairo_code
+fn main() {
+  let x: Option<Array<u128>> = Option::Some(array![1, 2, 3, 4, 5]);
+  if let Option::Some(v) = x {
+    v
+   } else {
+     array![]
+  };
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: This can be done in one call with `.unwrap_or_default()`
+ --> lib.cairo:4:3
+  |
+4 |     if let Option::Some(v) = x {
+  |  ___-
+5 | |     v
+6 | |    } else {
+7 | |      array![]
+8 | |   };
+  | |___-
+  |
+
+//! > fixed
+fn main() {
+  let x: Option<Array<u128>> = Option::Some(array![1, 2, 3, 4, 5]);
+  x.unwrap_or_default();
+}
+
+//! > ==========================================================================
+
+//! > manual unwrap or default for if let with comments
+
+//! > cairo_code
+fn main() {
+  let a: Option<ByteArray> = Option::Some("Helok");
+
+  if let Option::Some(v) = a {
+    // testing with comments
+    v
+   } else {
+    // testing with comments
+    Default::default()
+  };
+}
```

Comment:
if there are comments it should add a diagnostic

Reply:
It does.

from line 48 downwards?

I added two cases for comments, both with if let and match

Reply:
https://github.com/JoE11-y/cairo-lint/blob/main/crates/cairo-lint-core/tests/test_files/manual/manual_unwrap_or_default#L34